nixos/modules/virtualisation/glance.nix

```diff
@@ -0,0 +1,199 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.virtualisation.glance;
+  commonConf = ''
+    [database]
+    connection = mysql://glance:${cfg.dbPassword}@${cfg.dbHost}/glance
```

Comment:
It's the best when mysql client uses mysql client config files only. They allow `!include`, and thus you could have non-sensitive parameters in a "public" file and secret file included in it.